lib/jobs/pdu-node-relations.js

```diff
@@ -0,0 +1,142 @@
+    waterline,
+    Logger,
+    util,
+    Promise,
+    assert
+) {
+
+    var logger = Logger.initialize(pduRelationsJobFactory);
+
+    /**
+     *
+     * @param {Object} [options]
+     * @constructor
+     */
+    function PduRelationsJob(options, context, taskId) {
+        PduRelationsJob.super_.call(this, logger, options, context, taskId);
+
+        this.nodeId = context.target || options.nodeId;
+        assert.isMongoId(this.nodeId);
+    }
+
+    util.inherits(PduRelationsJob, BaseJob);
+
+    /**
+     * Globals
+     */
+
+    var RELATION_TYPE = 'poweredBy';
+
+    /**
+     * @memberOf PduRelationsJob
+     */
+    PduRelationsJob.prototype._run = function run() {
+        var self = this;
+        return waterline.nodes.findByIdentifier(self.nodeId)
+        .then(function (node) {
+            if (node.type !== 'pdu') {
+                throw new Error('Task runs only on PDU nodes');
+            }
+            var pduHost = _adjustIpStrFormat(node.snmpSettings.host);
+            return self._searchObmServices(pduHost);
+        })
+        .then(function () {
+            self._done();
+        })
+        .catch(function (err) {
+            self._done(err);
+        });
+
+    };
+
+    PduRelationsJob.prototype._searchObmServices =
+        function _searchObmServices(pduHost) {
+            var self = this;
+            return Promise.try(function () {
+                return waterline.obms.find();
+            })
+            .filter(function (obm) {
+                return (_validateIpAddressFormat(obm.config.host));
+            })
+            .map(function (result) {
+                //compare hosts
+                var obmServiceHost = _adjustIpStrFormat(result.config.host);
+                if (pduHost === obmServiceHost) {
+                    //find the node
+                    return waterline.nodes.findByIdentifier(result.node)
+                   .then(function (node) {
+                       if (!node) {
+                           throw new Error('Could not find node with identifier ' + result.node);
+                       }
+                       return node.relations;
+                   })
+                   .filter(function (entry) {
+                       return (entry.targets.indexOf(self.nodeId) < 0);
+                   })
+                   .then(function (relations) {
+                       relations.push({
+                           relationType: RELATION_TYPE,
+                           targets: [self.nodeId]
+                       });
+                       return waterline.nodes.updateByIdentifier(
+                           result.node,
+                           { relations: relations });
+                   });
+                }
+
+            });
+        };
+
+    /**
+     * verify ip address is 4 dec nbs separated by '.'
+     * returns a booleen
+     */
+    function _validateIpAddressFormat(ipString) {
+        var regex = /^(?:[0-9]{1,3}\.){3}[0-9]{1,3}$/;
```

Comment:
There is already a Constant for ipAddress: https://github.com/RackHD/on-core/blob/master/lib/common/constants.js#L215